### PR TITLE
test: add NIP-40 standalone expiration integration coverage

### PR DESCRIPTION
**Pull request by Priyanshubhartistm**
## Description
- Added NIP-40 standalone Gherkin scenarios in `test/integration/features/nip-40/nip-40.feature`.
- Added step definitions in `test/integration/features/nip-40/nip-40.feature.ts`.
- Scenario coverage added exactly as requested:
  - Kind 1 event with past expiration is not returned via REQ.
  - Kind 1 event with future expiration is returned via REQ.
  - Kind 1 event that expires after storage is not served to new subscribers.
- Step definitions use `EventTags.Expiration` and `EventExpirationTimeMetadataKey` from `src/constants/base.ts`.
- Added minimal test helper support in `test/integration/features/helpers.ts` to publish events without forcing expected OK success, so expiration edge cases can be asserted cleanly.
- Scope is test-only for this PR.

## Related Issue
- https://github.com/cameri/nostream/issues/502

## Motivation and Context
- NIP-33 already validates expiration behavior for parameterized replaceable events.
- There was no integration coverage for ordinary standalone events like kind 1 text notes.
- This closes that gap and codifies expected relay behavior for NIP-40 on non-replaceable events.

## How Has This Been Tested?
- Local pre-commit hook pipeline executed successfully:
  - lint
  - build:check
  - unit tests
- Integration suite command run:
  - `npm run docker:test:integration -- test/integration/features/nip-40/nip-40.feature`
- Observed behavior:
  - Scenario 1 passed.
  - Scenario 2 passed.
  - Scenario 3 currently fails against existing relay implementation, which is expected for a test-only PR that captures issue behavior.

## Screenshots (if appropriate):
- N/A

## Types of changes
- [x] Non-functional change (docs, style, minor refactor)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my code changes.
- [ ] I added a changeset, or this is docs-only and I added an empty changeset.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Priyanshubhartistm** [comment]
Hey @cameri, I resolved the merge conflicts and pushed the updated branch.
Please take another look when you have a moment.
thanku

File: src/handlers/subscribe-message-handler.ts [hunk 1]
```diff
@@ -74,6 +74,7 @@ export class SubscribeMessageHandler implements IMessageHandler, IAbortable {
       await pipeline(
         findEvents,
         streamFilter(propSatisfies(isNil, 'deleted_at')),
+        streamFilter(SubscribeMessageHandler.isNotExpired),
```
Comment (cameri) [inline src/handlers/subscribe-message-handler.ts]: isn't this the same as line 79?

Comment (cameri) [reply src/handlers/subscribe-message-handler.ts]: @kushagra0902 do you mind reviewing these changes?

Comment (Priyanshubhartistm) [reply src/handlers/subscribe-message-handler.ts]: @cameri Good catch, this was redundant after merge. I removed the duplicate pre-conversion expiration filter and kept a single expiration check in the replay pipeline. Tests are passing and the branch is updated.

File: test/integration/features/nip-40/nip-40.feature.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+) {
+  const events = this.parameters.events[author] as Event[]
+  const event = events[events.length - 1] as ExpiringEvent
+  const expirationTime = event[EventExpirationTimeMetadataKey]
+
+  expect(expirationTime).to.be.a('number')
+
+  const millisecondsUntilExpired = (Number(expirationTime) - now() + 1) * 1000
+
+  await wait(millisecondsUntilExpired)
```
Comment (cameri) [inline test/integration/features/nip-40/nip-40.feature.ts]: let's avoid using real time to test the integration, otherwise this slows down integration tests by a second.

instead, we should probably just seed the expired event directly into the database (with the expiration in the past), and then subscribe: we should not get the event.

or:
publish an expired event to the relay.
subscribe and we should not get it back.

**Review by cameri — APPROVED** [review]
